collections/_article/platform-state-in-csharp-for-godot-4-2.md

```diff
@@ -0,0 +1,99 @@
+
+Exporting to iOS is only supported from a MacOS device and requires using Xcode to build the final binaries.
+
+The official export template for the iOS simulator only supports the `x64` architecture.
+
+### Web
+
+The web platform is still **not supported** in 4.2, but progress has been made in loading GDExtensions which has some similarities.
+
+Godot uses [Emscripten](https://emscripten.org/) to compile C++ code to [Web Assembly](https://webassembly.org/) (WASM) that can be run as "native code" in web browsers. The Godot WASM acts as the main module and it's able to load other WASMs that are built as side modules, GDExtensions that are used in web exports are built as WASM side modules so Godot can load them dynamically.
+
+When building C# projects for the web platform, the built WASM expects .NET to be the main entry point and doesn't support [dynamic linking](https://github.com/dotnet/runtime/issues/75257). This is because, currently, the .NET runtime can only be built as a main module. So, unlike GDExtensions, the resulting WASM can't be loaded by Godot.
+
+### Desktop
+
+Desktop platforms have been supported since 4.0, but what some users may not know is that those platforms support all .NET runtimes. This is not new to 4.2 and, in fact, has been supported since 4.0. Godot's C# integration uses standard .NET so everything that works in a normal C# project should work in Godot too. To use a different runtime from the default (which is CoreCLR), users have to modify the C# project file (`.csproj`) as documented in the Microsoft documentation.
+
+To enable NativeAOT in desktop platforms make sure to target .NET 7.0 or higher and set the `<PublishAOT>` property to true. Also, since Godot uses reflection, make sure to root the assemblies (this is done automatically for you when exporting to iOS because NativeAOT is currently the only supported runtime). Here's an example C# project:
+
+```xml
+<Project Sdk="Godot.NET.Sdk/4.2.0">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <EnableDynamicLoading>true</EnableDynamicLoading>
+    <!-- Use NativeAOT. -->
+    <PublishAOT>true</PublishAOT>
+  </PropertyGroup>
+  <ItemGroup>
+    <!-- Root the assemblies to avoid trimming. -->
+    <TrimmerRootAssembly Include="GodotSharp" />
+    <TrimmerRootAssembly Include="$(TargetName)" />
+  </ItemGroup>
+</Project>
+```
+
+### Summary
+
+As a summary, the current platform support as of 4.2 is described in the table below:
+
+| Platform | Runtimes supported | Minimum required .NET version |
+| -: | - | - |
+| **Windows** | CoreCLR, Mono, NativeAOT | 6.0 (CoreCLR, Mono), 7.0 (NativeAOT) |
+| **macOS** | CoreCLR, Mono, NativeAOT | 6.0 (CoreCLR, Mono), 7.0 (NativeAOT) |
+| **Linux** | CoreCLR, Mono, NativeAOT | 6.0 (CoreCLR, Mono), 7.0 (NativeAOT) |
+| **Android** | Mono | 7.0 |
+| **iOS** | NativeAOT | 8.0 |
+| **Web** | - | - |
```

Comment:
The table doesn't look great :disappointed: 

![image](https://github.com/godotengine/godot-website/assets/3903059/4bc05da8-dd42-4b1e-9f48-aaa225a49734)

Reply:
We should add blog table styles in a dedicated PR.